**Dataset sobre areas protegidas - Gonzaléz Ciro**

In [1]:
#antes que nada instalamos todos los paquetes necesarios.
!pip install pandas plotly folium

**Primera parte - pequeño analisis del dataset**

In [2]:
import pandas as pd
area = pd.read_csv("area_protegida.csv")

que hicimos ahi? facil, importamos la libreria de pandas para poder leer archivos csv y definir una variable con la cual llamaremos mas facil al documento que en este caso es el de "area_protegida.csv".

**Lectura y primeros codigos para probar**

empezaremos usando codigos basicos para mostrar filas y columnas con el comando:"head".

In [3]:
area.head(3)

,lat,lng,gid,entidad,objeto,fna,gna,nam,tap,jap,caa,fdc,sag,area
0,-53.546655,-68.041297,362,falso,Área protegida,Sitio RAMSAR y Reserva Hemisférica Costa Atlán...,Sitio RAMSAR y Reserva Hemisférica,Costa Atlántica de Tierra del Fuego,2,2,17,SIB/IGN/Ley Provincial Nº 415/1998,IGN,762166
1,-31.404445,-64.357709,157,falso,Área protegida,Reserva Natural de la Defensa La Calera,Reserva Natural de la Defensa,La Calera,2,2,17,SIB/IGN/Protocolo adicional Nº 05/2009,IGN,136225
2,-32.245004,-60.734423,169,falso,Área protegida,Sitio Ramsar Delta del Parana,Sitio Ramsar,Delta del Parana,2,2,17,NaN,MAyDS,2430979


que hice ahi? mostre las primeras 3 filas y las columnas disponibles para buscar a detalle y seguir explorando o trabajando en este dataset.

In [4]:
area.shape

(395, 14)

aca podemos ver que con este comando o codigo nos da que el dataset actual contiene 395 filas y 14 columnas.

**Pruebas con estructuras y tipos de datos**

In [5]:
area.dtypes

lat        float64
lng        float64
gid          int64
entidad        str
objeto         str
fna            str
gna            str
nam            str
tap          int64
jap          int64
caa          int64
fdc            str
sag            str
area         int64
dtype: object

con este codigo podemos ver cuales son los tipos de datos que de vuelve un ejemplo seria "lat" o "lng" que contienen float64 que en pocas palabras siginifica que tiene valores en numeros decimales.

**Valores nulos**

In [6]:
area.isnull().sum()

lat         0
lng         0
gid         0
entidad     0
objeto      0
fna         0
gna         0
nam         0
tap         0
jap         0
caa         0
fdc        59
sag         1
area        0
dtype: int64

con este codigo podras ver y contar cuantos valores nulos o faltantes hay en cada columna del dataset.

**Segunda parte - Preguntas de analisis sobre areas protegidas**

In [7]:
import pandas as pd

In [8]:
import plotly.express as px

In [9]:
import folium

In [10]:
area = pd.read_csv("area_protegida.csv")

**1. ¿Cual es el tamaño total de la superficie protegida y cual es el promedio de cada area?**

si queremos saber cuanta tierra y agua esta protegida en total en el pais y cuanto mide en promedio una de estas areas podriamos usar el siguiente codigo.

In [11]:
area_total = area["area"].sum()
area_promedio = area["area"].mean()

print("superficie total protegida:", area_total)
print("superficie promedio de cada area:", area_promedio)

superficie total protegida: 491870721
superficie promedio de cada area: 1245242.3316455695


**2. ¿Cuales son las 5 categorias de proteccion que mas superficie protegen?**

en si hay categorias como "Parque Nacional" o "Reserva Provincial" asi que con el siguiente codigo podremos ver cuales de estas categorias suman mas terreno protegido en total.

In [12]:
top_categorias_area = area.groupby("gna")["area"].sum().nlargest(5)
print(top_categorias_area)

gna
Área Marina Protegida     114830836
Reserva de Biósfera        96634910
Sitio Ramsar               38042819
Parque Nacional            31702666
Área Natural Protegida     17750488
Name: area, dtype: int64


**3. ¿Cuantas areas protegidas tienen un tamaño menor a la media?**

la mitad de las areas del pais estan por debajo de la media en tamaño (79,300) con este codigo podremos contar cuantas son exactamente.

In [13]:
mediana_area = area["area"].median()
areas_pequenas = area[area["area"] < mediana_area]

print("la cantidad de areas mas pequeñas que la mitad de la tabla son:", len(areas_pequenas))

la cantidad de areas mas pequeñas que la mitad de la tabla son: 197


**4. ¿Que porcentaje de las areas no tiene cargado el documento legal (fdc)?**

Si nosotros quisieramos saber cuantas areas están "olvidadas" en el registro y no tienen cargada la ley o decreto que las creo para ver su legalidad actual podriamos usar el siguiente comando.

In [14]:
porcentaje_sin_ley = (area["fdc"].isna().mean() * 100).round(2)
print("porcentaje de areas sin el documento legal:", porcentaje_sin_ley, "%")

porcentaje de areas sin el documento legal: 14.94 %


**5. ¿Cual es la reserva mas pequeña de todas y cuanto mide?**

este codigo es por si queremos encontrar el area protegida mas chiquita de todo el mapa.

In [15]:
mas_chica = area.nsmallest(1, "area")[["nam", "gna", "area"]]
print(mas_chica)

              nam                gna  area
22  Vicente López  Reserva Ecológica    27


**6. ¿Cuantas "areas marinas protegidas" hay y cuanto territorio suman?**

con este codigo queremos ver cuantas de estas areas protegen especificamente el mar y el agua y que tan grandes son juntas.

In [16]:
marinas = area[area["gna"] == "area marina protegida"]
print("cantidad de areas marinas:", len(marinas))
print("suma de su superficie marina:", marinas["area"].sum())

cantidad de areas marinas: 0
suma de su superficie marina: 0


**7. ¿Cuantas areas maneja el instituto geografico nacional (IGN) frente al ministerio de ambiente (MAyDS)?**

si necesitamos ver el conteo de cuantas areas tiene bajo su control cada organismo del gobierno este codigo nos lo muestra.

In [17]:
conteo_organismos = area["sag"].value_counts()
print(conteo_organismos)

sag
IGN      342
MAyDS     52
Name: count, dtype: int64


**8. ¿Cuales son los nombres de las 3 areas con mayor superficie de la muestra?**

aca esta una lista rapida con los nombres de los 3 "gigantes" del mapa.

In [18]:
top3_nombres = area.nlargest(3, "area")[["nam", "area"]]
print(top3_nombres)

                           nam      area
40                     Yaganes  58234860
34  Namuncurá - Banco Burdwood  30740644
62              Patagonia Azul  30553073


**9. ¿donde estan ubicados los "parques nacionales"?**

en el mapa se ve claramente cada punto del pais en donde se puede ver el nombre de cada parque nacional.

In [19]:
import folium

parques = area[area["gna"] == "Parque Nacional"]
mapa_parques = folium.Map(location=[-38.41, -63.61], zoom_start=4)

for _, fila in parques.iterrows():
    folium.Marker(
        location=[fila["lat"], fila["lng"]],
        popup=fila["nam"]
    ).add_to(mapa_parques)
    
mapa_parques.save("mapa_parques.html")
mapa_parques


**10. ¿donde estan las 15 areas protegidas mas grandes?**

en el siguiente mapa se podran ver las ubicaciones de las 15 areas protegidas mas grandes.

In [20]:
import folium

grandes = area.nlargest(15, "area")
mapa_grandes = folium.Map(location=[-38.41, -63.61], zoom_start=4)

for _, fila in grandes.iterrows():
    folium.CircleMarker(
        location=[fila["lat"], fila["lng"]],
        radius=12,
        color="red",
        fill=True,
        fill_opacity=0.6,
        popup=f"{fila['nam']} (Tamaño: {fila['area']})"
    ).add_to(mapa_grandes)

mapa_grandes.save("mapa_grandes.html")
mapa_grandes
